python/mxnet/text/text.py

```diff
@@ -0,0 +1,125 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Provide text utilities."""
+from __future__ import absolute_import
+from __future__ import print_function
+
+from collections import Counter
+import logging
+import numpy as np
+import os
+import re
+
+from ..base import numeric_types
+from .. import ndarray as nd
+from .. import io
+
+
+def count_tokens_from_str(tokens, token_delim=" ", seq_delim="\n",
+                          to_lower=False):
```

Comment:
consider adding the counter as an optional argument, with the default value being an empty counter. this way, the same function can be used to either create new counter or update existing counter, which effectively removes the assumption of having to store a whole corpus in memory.

Reply:
resolved.